### PR TITLE
ENH: add a x2x_scan

### PR DESCRIPTION
**Pull request by tacaswell**
## Description

This is a generic version of a theta-2theta scan.

## Motivation and Context

Watching @ambarb do scans while testing the tardis at CSX she typed this signature into `relative_inner_product_scan` as a bit of wishful API.

@ambarb Did I understand this correctly and is this useful?  Do you have a better name?

## How Has This Been Tested?

Still needs test

### CODE REVIEW

**Comment by ambarb** [comment]
yes, that is correct.  the generic option in bluesky is more flexible because it lets you decide if the range of two theta is 2x theta.  spec's th2th assumes this is what you would like so you only need to input the range for 2theta.

**Comment by ambarb** [comment]
@tacaswell yes, after looking at the doc string your `x2x_scan` does what we discussed.  And I think your name for the scan makes good sense.  

However, the 2x motor is usually the detector and the frame of reference for the scientist creating the scan.  You may consider switching the order of the motors and the name.

For reference to discussion:
spec always did this a little backward as your scan arguments would be for the fast motor (and it would then move the second motor half as much) so the name for the scan (`th2th`) was the inverse of what was the underlying d2scan.
https://certif.com/spec_manual/mac_3_9_2.html

File: bluesky/plans.py [hunk 1]
```diff
@@ -1413,6 +1413,50 @@ def fly(flyers, *, md=None):
     yield from bps.close_run()
 
 
+def x2x_scan(detectors, motor1, motor2, start, stop, num, *,
+             per_step=None, md=None):
+    """
+    Relatively scan over two motors in a 1:2 ratio
+
+    This is a pasteurized version of a theta2theta scan
```
Comment (danielballan) [inline bluesky/plans.py]: As much as I love metaphorical language, "generalized" might be friendlier here.

**Comment by tacaswell** [comment]
Should this be flipped to move the second motor half of the first?

File: bluesky/plans.py (hunk 1 above)
Comment (tacaswell) [reply bluesky/plans.py]: I suspect that is me losing to the spell-checker....

**Comment by tacaswell** [comment]
flipped it so that the second motor moves half as much (as `2xx_scan` is not a valid function name in python).

**Comment by danielballan** [comment]
I'm comfortable with the name `x2x_scan` even though its slightly misleading. I think most users of this plan will be reaching for it because they know about `theta2theta` and are accustomed to SPEC's (arguably backward) convention.

**Comment by danielballan** [comment]
I think this is good to merge but I'd like @ambarb to take a last look and sign off before we merge it.

**Comment by tacaswell** [comment]
and it needs a test!

**Comment by ambarb** [comment]
Do you guys want someone from the beamline side to test it?

**Comment by tacaswell** [comment]
@awalter-bnl Can I had this off to you?   Can you

 - make sure the name and signature make sense and match your expectations
 - add a test (something like listify it and assert that the sets have the right relationships)

You should be able to push directly to my branch.

**Comment by awalter-bnl** [comment]
Hi @tacaswell , I can take a look at this. Can you provide some direction as to where the tests are usually kept? This way I can follow the existing layout for this new test.

**Comment by danielballan** [comment]
The tests of plans are a little scattered about for historical reasons, but [`test_plans.py`](https://github.com/NSLS-II/bluesky/blob/master/bluesky/tests/test_plans.py#L184) is a good place to put new ones.

Have you encountered pytest _fixtures_ before? Notice that the tests take `RE` and `hw` ("hardware") as arguments. When pytest runs the tests, it inspects the tests' function signatures and automatically produces a RunEngine instance `RE` and a `SimpleNamespace` (a generic, dottable object) `hw` for use in the test so that we don't have to type that boilerplate every time or update it in hundreds of different places when something changes. Something like `hw.motor` and `hw.det` are appropriate in this case.

Also notice the usage `MsgCollector`, which is useful for comparing that _actual_ messages to the expected ones. That is worth imitating. But no need to imitate @tacaswell's spelling of 'expected' as 'expecte'. #thedangersofautocomplete ;- )

**Comment by awalter-bnl** [comment]
Thanks @danielballan I had a 'brief' introduction when doing the 'spiral_square' plan, but Maksim did most of the test function work. I will ask questions if I have any !-)

**Comment by mrakitin** [comment]
For reference: https://github.com/NSLS-II/bluesky/pull/1001/files#diff-7d8393473493d7106f01ce5900958096R554

**Comment by awalter-bnl** [comment]
@tacaswell I have checked as requested, and added the test to tests/test_scans.py (including running it to ensure there are no issues). I don't see any issues with the name or signature.

Take a look at the test and let me know if you see any issues.

File: bluesky/tests/test_scans.py [hunk 2]
```diff
@@ -479,3 +479,22 @@ def test_relative_fermat_spiral(RE, hw):
     approx_multi_traj_checker(RE, scan,
                               _get_fermat_data(start_x, start_y),
                               decimal=2)
+
+
+def test_x2x_scan(RE,hw):
```
Comment (danielballan) [inline bluesky/tests/test_scans.py]: Style thing: needs a space after this comma. (Soon we'll probably apply a linter, as many large project do, so these kinds of things we'll actually be _tested_.)

**Comment by danielballan** [comment]
Thanks, @awalter-bnl and @mrakitin!

**Comment by awalter-bnl** [comment]
Thanks @danielballan, I missed that space but it has now been fixed.

File: bluesky/tests/test_scans.py (hunk 2 above)
Comment (danielballan) [reply bluesky/tests/test_scans.py]: Yes, there is a lot of cleaning up to do on all of our projects. I have a branch (somewhere...) that attempted to do this for bluesky but it's probably hopelessly out of date now. Might as well start over.

**Pull request merged**